### PR TITLE
[BACK-2489] Offset Period support with Deltas

### PR DESCRIPTION
**Pull request by Roukoswarf**

### CODE REVIEW

File: data/summary/types/bgm.go
Comment (tjotala) [inline data/summary/types/bgm.go:230]: Is this missing a delta for `GlucoseManagementIndicator`? Is that meaningful?

Comment (tjotala) [reply data/summary/types/bgm.go:230]: Ugh, nevermind. No such thing for BGM.

File: data/summary/types/test/summary.go
Comment (tjotala) [inline data/summary/types/test/summary.go:257]: Ditto here

File: data/summary/types/test/summary.go
Comment (Roukoswarf) [reply data/summary/types/test/summary.go:99]: It was moved to the top of this area as it did not match the ordering of the variables that follow.

File: data/summary/types/test/summary.go
Comment (Roukoswarf) [reply data/summary/types/test/summary.go:257]: It was moved to the top of this area as it did not match the ordering of the following variables.

**Review by tjotala — COMMENTED** [review]
Someone else with more current working experience in this repo & Golang should review this. Tagged Todd, but not limited to him.

File: data/summary/types/bgm.go
Comment (lostlevels) [inline data/summary/types/bgm.go:88]: Is this commented out for a reason or can it be deleted?

File: data/summary/types/bgm.go
Comment (lostlevels) [inline data/summary/types/bgm.go:102]: Same here with commented out

File: data/summary/types/cgm.go
Comment (lostlevels) [inline data/summary/types/cgm.go:135]: Is this commented out code still needed?

File: data/summary/types/cgm.go
Comment (lostlevels) [inline data/summary/types/cgm.go:149]: same with this comment

**Review by lostlevels — DISMISSED** [review]
Looks fine minus various commented out code. Not sure if they are for the future or should be removed.

**Comment by toddkazakov** [comment]
I agree with @lostlevels. All commented code should be removed.

**Comment by Roukoswarf** [comment]
I will stash the comments in a branch locally so i dont lose them, as they represent a possible but now unlikely path forward and remove them from the PR.

**Review by lostlevels — DISMISSED** [review]

**Review by toddkazakov — COMMENTED** [review]
The changes to the summary structure are not backward compatible with clinic-worker (thus not compatible with the EHR integration). Please make sure you have corresponding PRs that are ready to be merged before merging this one.

File: data/summary/types/bgm.go
Comment (Roukoswarf) [reply data/summary/types/bgm.go:258]: yes, copy paste error, fixed it in the CGM stats, but missed it here, thank you.

**Review by toddkazakov — APPROVED** [review]

**Pull request merged**
